File: Neetcode Questions/Math & Geometry/plus-one.cpp

// https://leetcode.com/problems/plus-one/

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {

        int carry = 1;
        for(int i = digits.size()-1; i >= 0; i--){
            int currSum = digits[i] + carry;
            carry = currSum/10;
            digits[i] = currSum%10;
        }

        if(carry > 0) digits.insert(digits.begin(), carry);

        return digits;
    }
};

File: Neetcode-150-js/Math & Geometry/plus-one.js

// https://leetcode.com/problems/plus-one/description/

/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let carry = 1;
    let len = digits.length;
    let i = len-1;
    const ans = Array.from({length : len}, () => 0);
    while(i >= 0){
        let sum = digits[i] + carry;
        ans[i] = sum%10;
        carry = Math.floor(sum/10);
        i--;
    }
    if(carry > 0) ans.unshift(carry);
    return ans;
};